extide/gradle/src/org/netbeans/modules/gradle/GradleProjectConnection.java

```diff
@@ -111,6 +111,10 @@ public synchronized void close() {
         compatConn = null;
     }
 
+    boolean hasConnection() {
```

Comment:
If this class is meant to be thread-safe, which seems to be the case, then this method should be synchronized. Though it probably doesn't matter for the curent use, which is just for the single test class.